src/ariel-os-rt/build.rs

```diff
@@ -21,21 +21,22 @@ fn main() {
     // Put the linker scripts somewhere the linker can find them
     let out = &PathBuf::from(env::var_os("OUT_DIR").unwrap());
 
-    if let Some(context) = context_any(&["cortex-m", "riscv"]) {
-        let insert_before = match *context {
-            "riscv" => ".trap",
-            "cortex-m" => ".data",
-            _ => unreachable!(),
+    if let Some(context) = context_any(&["cortex-m", "riscv", "xtensa"]) {
+        let insert_somewhere = match *context {
```

Comment:
This now contains the complete line, making it more flexible. e.g., empty string for xtensa. We can also do `INSERT BEFORE X;` now.